app/src/main/java/com/babylon/wallet/android/presentation/history/HistoryViewModel.kt

```diff
@@ -0,0 +1,420 @@
+package com.babylon.wallet.android.presentation.history
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.babylon.wallet.android.domain.model.HistoryFilters
+import com.babylon.wallet.android.domain.model.Selectable
+import com.babylon.wallet.android.domain.model.TransactionClass
+import com.babylon.wallet.android.domain.model.TransactionHistoryData
+import com.babylon.wallet.android.domain.model.TransactionHistoryItem
+import com.babylon.wallet.android.domain.model.assets.AccountWithAssets
+import com.babylon.wallet.android.domain.model.resources.Resource
+import com.babylon.wallet.android.domain.usecases.assets.GetAccountFirstTransactionDateUseCase
+import com.babylon.wallet.android.domain.usecases.assets.GetAccountHistoryUseCase
+import com.babylon.wallet.android.domain.usecases.assets.GetWalletAssetsUseCase
+import com.babylon.wallet.android.presentation.common.OneOffEvent
+import com.babylon.wallet.android.presentation.common.OneOffEventHandler
+import com.babylon.wallet.android.presentation.common.OneOffEventHandlerImpl
+import com.babylon.wallet.android.presentation.common.StateViewModel
+import com.babylon.wallet.android.presentation.common.UiMessage
+import com.babylon.wallet.android.presentation.common.UiState
+import com.babylon.wallet.android.utils.toMonthString
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import rdx.works.profile.data.model.apppreferences.Radix.dashboardUrl
+import rdx.works.profile.derivation.model.NetworkId
+import rdx.works.profile.domain.GetProfileUseCase
+import rdx.works.profile.domain.accountOnCurrentNetwork
+import timber.log.Timber
+import java.time.Instant
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.temporal.TemporalAdjusters
+import javax.inject.Inject
+
+@Suppress("LongParameterList", "TooManyFunctions")
```

Comment:
why "LongParameterList"? you have only 5, you are inside the threshold which is good 👌🏽